File: Java/Lamacz gg.java
import java.util.Map;
import java.util.HashMap;
import java.util.Scanner;

public class Main
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        Map<Character, Integer> firstLetterValue = new HashMap<>();
        Map<Character, Integer> secondLetterValue = new HashMap<>();
        for (char c = 'A'; c <= 'P'; c++) 
        {
            firstLetterValue.put(c, c - 'A'); 
            secondLetterValue.put(c, (c - 'A') * 16);
        }
        while (input.hasNextLine()) 
        {
            String encodedPassword = input.nextLine();
            StringBuilder decodedPassword = new StringBuilder();
            for (int i = 0; i < encodedPassword.length(); i += 2) 
            {
                char firstChar = encodedPassword.charAt(i);
                char secondChar = encodedPassword.charAt(i + 1);
                int first = firstLetterValue.get(firstChar);
                int second = secondLetterValue.get(secondChar);
                char decodedChar = (char) (first + second);
                decodedPassword.append(decodedChar);
            }
            System.out.println(decodedPassword);
        }
        input.close();
    }
}

File: C++/Lamacz gg.cpp
#include <map>
#include <string>
#include <iostream>

int main() 
{
    std::string encoded_password;
    std::map<char, int> first_letter_value;
    std::map<char, int> second_letter_value;
    for (char c = 'A'; c <= 'P'; ++c) 
    {
        first_letter_value[c] = c - 'A';  
        second_letter_value[c] = (c - 'A') * 16; 
    }
    while (std::cin >> encoded_password)
    {
        std::string decoded_password = "";
        for (size_t i = 0; i < encoded_password.length(); i += 2) 
        {
            int first = first_letter_value[encoded_password[i]];
            int second = second_letter_value[encoded_password[i + 1]];
            char decoded_char = static_cast<char>(first + second);
            decoded_password += decoded_char;
        }
        std::cout << decoded_password << '\n';
    }
    return 0;
}